x-pack/plugins/ml/public/application/services/ml_api_service/index.ts

```diff
@@ -32,7 +32,11 @@ import {
   FieldHistogramRequestConfig,
   FieldRequestConfig,
 } from '../../datavisualizer/index_based/common';
-import { DataRecognizerConfigResponse, Module } from '../../../../common/types/modules';
+import {
+  DatafeedOverride,
```

Comment:
`DatafeedOverride` is for overriding the datafeed config when calling the module setup and so i don't think should be used here.
It looks like the standard `Datafeed` interface would be better suited throughout this PR.